src/lightning/pytorch/callbacks/progress/rich_progress.py

```diff
@@ -167,11 +167,13 @@ def render(self, task: "Task") -> Text:
             if self._trainer.training and task.id != self._current_task_id:
                 return self._tasks[task.id]
 
-            text = ""
-            for k, v in self._metrics.items():
-                text += f"{k}: {round(v, 3) if isinstance(v, float) else v} "
+            text = " ".join(self._generate_metrics_texts())
             return Text(text, justify="left", style=self._style)
 
+        def _generate_metrics_texts(self):
```

Comment:
```suggestion
        def _generate_metrics_texts(self) -> Generator:
```
(needs import from typing)

Reply:
Thank you, added it